### ISSUE TITLE
AWS healthcheck error 

### ISSUE THREAD

**Issue by k-o-ta** [privacysandbox/protected-auction-key-value-service#9]
Hi

I tried to test key-value service on AWS after solving these issues(https://github.com/privacysandbox/fledge-key-value-service/issues/7, https://github.com/privacysandbox/fledge-key-value-service/issues/8), and got an error on AWS EC2 target groups health check.
<img width="1207" alt="key-value-service-halthcheck" src="https://user-images.githubusercontent.com/5237032/226091733-f02f2ca0-644e-49ac-a0d2-e7eaa7b70f21.png">

I suspected that `-p` option is needed [here](https://github.com/privacysandbox/fledge-key-value-service/blob/f01c07ae71bec764c193c53d43facda275ae7a2d/production/terraform/aws/services/autoscaling/instance_init_script.tftpl#L38-L43), and restart docker container like below.
```
# on local 
mssh instance_id_of_ssh_instance  --region us-east-1              

# on ssh instance
mssh instance_id_of_instance  --region us-east-1

# on instance
# healthcheck
$ curl --verbose "http://localhost:51052/healthcheck"
*   Trying 127.0.0.1:51052...
* connect to 127.0.0.1 port 51052 failed: Connection refused
*   Trying [::1]:51052...
* connect to ::1 port 51052 failed: Connection refused
* Failed to connect to localhost port 51052 after 0 ms: Couldn't connect to server
* Closing connection 0
curl: (7) Failed to connect to localhost port 51052 after 0 ms: Couldn't connect to server

# docker stop
docker stop container_id

# docker run with port forwarding option
docker run --rm -d -p 51052:51052 -p 9902:9902 -v /etc/envoy/envoy.yaml:/etc/envoy/envoy.yaml envoyproxy/envoy-distroless:v1.24.1 --config-path /etc/envoy/envoy.yaml -l warn

# healthcheck retry
$ curl --verbose "http://localhost:51052/healthcheck"
*   Trying 127.0.0.1:51052...
* Connected to localhost (127.0.0.1) port 51052 (#0)
> GET /healthcheck HTTP/1.1
> Host: localhost:51052
> User-Agent: curl/7.88.1
> Accept: */*
> 
* Recv failure: Connection reset by peer
* Closing connection 0
curl: (56) Recv failure: Connection reset by peer
```

It changed the error message, but the health check still remained failed.

**Comment by peiwenhu** [comment]
Hi, thanks for raising this. We suspect that this is the same thing we also discovered recently and our fix will be in the coming release. We're working on getting the release out now

**Comment by peiwenhu** [comment]
Hi, we have released v0.8.0. We believe that this problem is addressed as part of it. Please let me know if you still run into it. thanks

**Comment by k-o-ta** [comment]
I have tried 0.8.0 and confirmed that the problem has been resolved. Thank you!
